### PR TITLE
Add sparse benchmarks for AbstractUnitRange

### PR DESCRIPTION
**Pull request by timholy**
Ref https://github.com/JuliaLang/julia/pull/22175

### CODE REVIEW

**Comment by tkelman** [comment]
guess we need to make these conditional on `isdefined(Base, :OneTo)` ?

File: appveyor.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ environment:
   - JULIAVERSION: "julialang/bin/winnt/x64/0.4/julia-0.4-latest-win64.exe"
   - JULIAVERSION: "julialang/bin/winnt/x86/0.5/julia-0.5-latest-win32.exe"
   - JULIAVERSION: "julialang/bin/winnt/x64/0.5/julia-0.5-latest-win64.exe"
+  - JULIAVERSION: "julialang/bin/winnt/x86/0.6/julia-0.5-latest-win32.exe"
+  - JULIAVERSION: "julialang/bin/winnt/x64/0.6/julia-0.5-latest-win64.exe"
```
Comment (Evizero) [inline appveyor.yml]: should these links be updated?

Comment (Evizero) [inline appveyor.yml]: (aside from the typo mixing 0.6 with 0.5 in one line)

Comment (timholy) [inline appveyor.yml]: Good catch

File: appveyor.yml [hunk 2]
```diff
@@ -21,7 +23,7 @@ notifications:
 install:
```
Comment (Evizero) [inline appveyor.yml]: One more tiny nitpick. This line will be needed for https I think

```
 - ps: "[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12"
```

Comment (timholy) [inline appveyor.yml]: Thanks. Added.

**Pull request merged**